src/app/utils/utils.js

```diff
@@ -265,16 +240,21 @@ export const extractRevisionNumber = (charmName) => charmName.split("-").pop();
 
 /**
   Returns a link to the charm icon for the provided charm name.
-  @param {String} namespace The fully qualified charm name.
+  @param {String} charmId The fully qualified charm name.
   @returns {String} The link to the charm icon.
 */
-export const generateIconPath = (namespace) => {
-  if (namespace.indexOf("local:") === 0) {
+export const generateIconPath = (charmId) => {
+  if (charmId.indexOf("local:") === 0) {
     return defaultCharmIcon;
   }
-  if (namespace.indexOf("cs:") === 0) {
-    namespace = namespace.replace("cs:", "");
-    return `https://api.jujucharms.com/charmstore/v5/${namespace}/icon.svg`;
+  if (charmId.indexOf("cs:") === 0) {
+    charmId = charmId.replace("cs:", "");
+    return `https://api.jujucharms.com/charmstore/v5/${charmId}/icon.svg`;
+  }
+  if (charmId.indexOf("ch:") === 0) {
```

Comment:
All the lines of code changed outside of this conditional are a driveby to clean-up the code.